File: Sales_Recon/src/dplyr/1-2-1_List_of_price_download.R
library(dplyr)
library(readr)
library(stringr)
library(readxl)
library(janitor)
library(here)
library(glue)


# Path
path <- here("Sales_Recon")


# Functions
main <- function() {
  # Variables
  source(here(path, "src", "dplyr", "common_variable.R"))

  # Filenames
  input_1 <- here(
    path, "data", "SAP", glue("Billing_0180_{year}_{month}.xlsx")
  )
  input_2 <- here(
    path, "data", "SAP", glue("Billing_2182_{year}_{month}.xlsx")
  )
  output_file <- here(path, "output", "1-2-1. list of price download.csv")

  # Read data
  df_0180 <- read_excel(input_1) |> clean_names(ascii = FALSE)
  df_2182 <- read_excel(input_2) |> clean_names(ascii = FALSE)

  # Process data
  df <- bind_rows(df_0180, df_2182)

  df <- df |>
    select(c("material_number")) |>
    distinct()

  # Write data
  write_csv(df, output_file, na = "")
  print("A file is created")
}

main()

File: Sales_Recon/src/pandas/1-2-1_List_of_price_download.py
import pandas as pd
from pathlib import Path
from janitor import clean_names


# Path
path = Path(__file__).parent.parent.parent


# Functions
def main():

    # Variables
    from common_variable import year, month

    # Filenames
    input_1 = path / "data" / "SAP" / f"Billing_0180_{year}_{month}.xlsx"
    input_2 = path / "data" / "SAP" / f"Billing_2182_{year}_{month}.xlsx"
    output_file = path / "output" / "1-2-1. list of price download_test.csv"

    # Read data
    df_0180 = pd.read_excel(input_1).clean_names(strip_accents=False)
    df_2182 = pd.read_excel(input_2).clean_names(strip_accents=False)

    # Process data
    df = (
        pd.concat([df_0180, df_2182])
        .loc[:,"material_number"]  # select column
        .drop_duplicates()
    )

    # Write data
    df.to_csv(output_file, index=False)
    print("A file is created")


if __name__ == "__main__":
    main()
